drivers/wifi/siwx91x/siwx91x_wifi.c

```diff
@@ -1039,6 +1042,224 @@ static int siwx91x_get_version(const struct device *dev, struct wifi_version *pa
 	return 0;
 }
 
+static void siwx91x_set_ps_param_wakeup_mode(struct wifi_ps_params *params,
+					     sl_wifi_performance_profile_t *sl_ps_profile)
+{
+	if (params->wakeup_mode == WIFI_PS_WAKEUP_MODE_DTIM) {
+		sl_ps_profile->dtim_aligned_type = 1;
+		sl_ps_profile->listen_interval = 0;
+	} else if (params->wakeup_mode == WIFI_PS_WAKEUP_MODE_LISTEN_INTERVAL) {
+		sl_ps_profile->dtim_aligned_type = 0;
+	} else {
+		/* No action needed */
+	}
+}
+
+static int siwx91x_set_ps_param_timeout(sl_wifi_performance_profile_t *sl_ps_profile,
+					   struct wifi_ps_params *params)
+{
+	if (params->timeout_ms < SLI_DEFAULT_MONITOR_INTERVAL ||
+	    params->timeout_ms > SIWX91X_PS_MAX_MONITOR_INTERVAL_MS) {
+		params->fail_reason = WIFI_PS_PARAM_FAIL_CMD_EXEC_FAIL;
+		return -EINVAL;
+	}
+
+	sl_ps_profile->monitor_interval = params->timeout_ms;
+	return 0;
+}
+
+static int siwx91x_set_ps_param_exit_strategy(struct siwx91x_dev *sidev,
+						 struct wifi_ps_params *params,
+						 sl_wifi_performance_profile_t *sl_ps_profile)
+{
+	if (params->exit_strategy == WIFI_PS_EXIT_EVERY_TIM) {
+		if (sidev->exit_strategy_mode == ASSOCIATED_POWER_SAVE_LOW_LATENCY) {
+			return 0;
+		}
+
+		sidev->exit_strategy_mode = ASSOCIATED_POWER_SAVE_LOW_LATENCY;
+	} else if (params->exit_strategy == WIFI_PS_EXIT_CUSTOM_ALGO) {
+		if (sidev->exit_strategy_mode == ASSOCIATED_POWER_SAVE) {
+			return 0;
+		}
+
+		sidev->exit_strategy_mode = ASSOCIATED_POWER_SAVE;
+	} else {
+		params->fail_reason = WIFI_PS_PARAM_FAIL_INVALID_EXIT_STRATEGY;
+		return -EINVAL;
+	}
+
+	if (sl_ps_profile->profile == HIGH_PERFORMANCE) {
+		/* Exit strategy takes effect when power save is enabled */
+		return 0;
+	}
+
+	sl_ps_profile->profile = sidev->exit_strategy_mode;
+
+	/* Apply to power save settings */
+	return SIWX91X_SET_PS;
+}
+
+static int siwx91x_set_ps_profile(const struct device *dev,
+				  sl_wifi_performance_profile_t *sl_ps_profile, enum wifi_ps state)
+{
+	struct siwx91x_dev *sidev = dev->data;
+
+	switch (state) {
+	case WIFI_PS_ENABLED:
+		if (sl_ps_profile->profile != HIGH_PERFORMANCE) {
+			return -EALREADY;
+		}
+
+		/* Use configured exit strategy, fallback to Fast PSP if not set */
+		if (sidev->exit_strategy_mode == HIGH_PERFORMANCE) {
+			sl_ps_profile->profile = ASSOCIATED_POWER_SAVE_LOW_LATENCY;
+		} else {
+			sl_ps_profile->profile = sidev->exit_strategy_mode;
```

Comment:
If you change the default value of `exit_strategy_mode` into `ASSOCIATED_POWER_SAVE_LOW_LATENCY`, you can simplify the statement above.

Then `exit_strategy_mode` will only contains `ASSOCIATED_POWER_SAVE_LOW_LATENCY` or `ASSOCIATED_POWER_SAVE`, so it will be easier to understand.